pkg/odo/cli/component/create.go

```diff
@@ -464,11 +464,17 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 
 	// this populates the LocalConfigInfo as well
 	co.Context = genericclioptions.NewContextCreatingAppIfNeeded(cmd)
-
 	if err != nil {
 		return errors.Wrap(err, "failed intiating local config")
 	}
 
+	// Do not execute S2I specific code on Kubernetes Cluster
+	// return from here, if it is not an openshift cluster.
+	openshiftCluster, _ := co.Client.IsRouteSupported()
```

Comment:
I know it's not incorrect because k8s doesn't support routes, but we're hitting the bug because k8s doesn't support s2i while OpenShift does. Would it be a better idea to check if the backend cluster supports s2i instead of route?

This might even be a nitpick. :)

Reply:
I was also skeptical about this , we could check for imagestreams resource, would that be fine??

Reply:
Yes, I think that should be a good idea. :+1:

Reply:
Done.